base/applications/rapps/appview.cpp

```diff
@@ -1892,6 +1892,38 @@ CApplicationView::AppendTabOrderWindow(int Direction, ATL::CSimpleArray<HWND> &T
     m_AppsInfo->AppendTabOrderWindow(Direction, TabOrderList);
 }
 
+VOID
+CApplicationView::GetRestoreListSelectionData(LVITEMW &Item, WCHAR *Name, UINT NameLen)
+{
+    Item.mask = LVIF_TEXT|LVIF_STATE;
+    Item.iItem = -1, Item.iSubItem = 0;
+    Item.stateMask = LVIS_FOCUSED|LVIS_SELECTED;
+    Item.pszText = Name, Item.cchTextMax = NameLen;
+
+    HWND hList = m_ListView ? m_ListView->m_hWnd : NULL;
```

Comment:
How is it possible that the `m_ListView` could be NULL but this GetRestoreListSelectionData function be able to be called?
Wouldn't it be better if somehow it cannot be called from the user (i.e. F5 functionality be disabled) as long as the list view does not exist?
In case it's possible to improve the code in that regard, then I guess one could just straightforwardly use the `m_ListView` (or ASSERT for its existence).

Reply:
CMainWindow does not know the inner workings of CApplicationView and I'd rather it not crash for what is just a "nice to have" feature. It does seem like most of the code in CApplicationView just assumes that m_ListView is valid which begs the question of why it is a pointer in the first place.

Do you want me to just ASSERT instead?